cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -104,21 +98,18 @@
  * Workflow node state is updated whenever program state is updated
  * and we notice that the program belongs to a workflow.
  */
-public class AppMetadataStore extends MetadataStoreDataset {
+public class AppMetadataStore {
 
   public static final DatasetId APP_META_INSTANCE_ID = NamespaceId.SYSTEM.dataset(Constants.AppMetaStore.TABLE);
 
   private static final Logger LOG = LoggerFactory.getLogger(AppMetadataStore.class);
   private static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(new GsonBuilder()).create();
   private static final Type MAP_STRING_STRING_TYPE = new TypeToken<Map<String, String>>() { }.getType();
-  private static final Type BYTE_TYPE = new TypeToken<byte[]>() { }.getType();
-  private static final byte[] APP_VERSION_UPGRADE_KEY = Bytes.toBytes("version.default.store");
-  private static final byte[] RUN_COUNT_FIRST_UPGRADE_TIME = Bytes.toBytes("run.count.first.upgrade.time");
+  private static final String APP_VERSION_UPGRADE_KEY = "version.default.store";
```

Comment:
do we need this? Do we need to support upgrade? I think this is a relic from CDAP 2.6?

Reply:
Removed upgrade code

Reply:
I think old code here